### PR TITLE
Support for the 'schedule' event

### PR DESCRIPTION
**Pull request by anarast**
- Adds handling for the schedule event by getting the repo name/owner from the `GITHUB_REPOSITORY` env var and the ref from the `GITHUB_REF` env var
- Manually tested [here](https://github.com/anarast/autoupdate-test/actions/runs/765797693)

### CODE REVIEW

File: src/autoupdater.ts [hunk 1]
```diff
@@ -55,6 +60,34 @@ export class AutoUpdater {
...
+    if (ref === undefined) {
+      ghCore.error('GITHUB_REF value is undefined.');
+      return 0;
+    }
+
+    const splitRepoName = ownerAndRepo.split('/');
+    const repoOwner = splitRepoName[0];
+    const repoName = splitRepoName[1];
+
+    if (repoOwner === undefined || repoName === undefined) {
```
Comment (chinthakagodawita) [inline src/autoupdater.ts:81]: `repoOwner` will never be undefined, maybe a length check is better here? i.e. 
```suggestion
    if (splitRepoName.length !== 2) {
```

File: src/autoupdater.ts [hunk 2]
```diff
@@ -55,6 +60,34 @@ export class AutoUpdater {
     return isUpdated;
   }
 
+  async handleSchedule(): Promise<number> {
+    const ref = this.config.getValue('GITHUB_REF');
+    const ownerAndRepo = this.config.getValue('GITHUB_REPOSITORY');
```
Comment (chinthakagodawita) [inline src/autoupdater.ts:65]: What's the reason for not making these their own functions on the config loader?

Also - if you provided `true` as the second argument to `getValue`, the config loader will automatically throw an error for you (and you can remove the two checks below)

**Review by chinthakagodawita — CHANGES_REQUESTED** [review]
Could you update the README too?

File: src/autoupdater.ts (hunk 2 above)
Comment (anarast) [reply src/autoupdater.ts:65]: No particular reason - I'll add them as their own config functions

File: README.md [hunk 3]
```diff
@@ -79,15 +85,15 @@ jobs:
...
-          MERGE_CONFLICT_ACTION: "fail"
+          GITHUB_TOKEN: '${{ secrets.GITHUB_TOKEN }}'
+          DRY_RUN: 'false'
+          PR_FILTER: 'labelled'
+          PR_LABELS: 'autoupdate,keep up-to-date,integration'
+          EXCLUDED_LABELS: 'dependencies,wontfix'
+          MERGE_MSG: 'Branch was auto-updated.'
+          RETRY_COUNT: '5'
+          RETRY_SLEEP: '300'
+          MERGE_CONFLICT_ACTION: 'fail'
```
Comment (chinthakagodawita) [inline README.md:96]: Could you revert this change? I'd prefer to keep the double quotes in here

File: README.md [hunk 4]
```diff
@@ -22,49 +25,52 @@ jobs:
...
 For more information on customising event triggers, see [Github's documentation](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#push-event-push).
 
 The following events are supported:
-* push
-* pull_request
-* workflow_run
+
+- push
+- pull_request
+- workflow_run
```
Comment (chinthakagodawita) [inline README.md:39]: \+ schedule here?

File: test/autoupdate.test.ts [hunk 5]
```diff
@@ -513,6 +517,81 @@ describe('test `handlePush`', () => {
...
+      .reply(200, pullsMock);
+
+    const updated = await updater.handleSchedule();
+
+    expect(updated).toEqual(expectedPulls);
+    expect(updateSpy).toHaveBeenCalledTimes(expectedPulls);
+    expect(scope.isDone()).toEqual(true);
+  });
+
+  test('schedule event with undefined GITHIB_REPOSITORY env var', async () => {
```
Comment (chinthakagodawita) [inline test/autoupdate.test.ts:557]: 😄 🚫 🥝 
```suggestion
  test('schedule event with undefined GITHUB_REPOSITORY env var', async () => {
```

File: test/autoupdate.test.ts [hunk 6]
```diff
@@ -513,6 +517,81 @@ describe('test `handlePush`', () => {
...
+      .spyOn(config, 'githubRef')
+      .mockImplementation(() => `refs/heads/${base}`);
+
+    const event = dummyScheduleEvent;
+    const updater = new AutoUpdater(config, (event as unknown) as WebhookEvent);
+
+    await expect(updater.handleSchedule()).rejects.toThrowError();
+  });
+
+  test('schedule event with undefined GITHIB_REF env var', async () => {
```
Comment (chinthakagodawita) [inline test/autoupdate.test.ts:568]:
```suggestion
  test('schedule event with undefined GITHUB_REF env var', async () => {
```

File: src/router.ts [hunk 7]
```diff
@@ -22,9 +22,11 @@ export class Router {
       await this.updater.handlePush();
     } else if (eventName === 'workflow_run') {
       await this.updater.handleWorkflowRun();
+    } else if (eventName === 'schedule') {
+      await this.updater.handleSchedule();
     } else {
       throw new Error(
-        `Unknown event type '${eventName}', only 'push', 'pull_request' and 'workflow_run' are supported.`,
+        `Unknown event type '${eventName}', only 'push', 'pull_request', 'workflow_run', and 'schedule' are supported.`,
```
Comment (chinthakagodawita) [inline src/router.ts:29]: A thought: can `workflow_run` trigger `schedule` events? If so - do we need to update the workflow run handling?

File: src/autoupdater.ts [hunk 8]
```diff
@@ -79,10 +103,20 @@ export class AutoUpdater {
...
+      repository.owner.name,
+    );
   }
 
-  async pulls(ref: string, repository: Repository): Promise<number> {
+  async pulls(
+    ref: string,
+    repoName: string,
+    repoOwnerLogin: string,
+    repoOwnerName?: string,
```
Comment (chinthakagodawita) [inline src/autoupdater.ts:118]: Could you add a simple check to make sure these are valid? e.g.
```typescript
const owner = repoOwnerName ?? repoOwnerLogin;
if (owner.length === 0) {
  ghCore.error('Invalid repository owner provided');
  return 0;
}
if (repoName.length === 0) {
  ghCore.error('Invalid repository name provided');
  return 0;
}
```

**Review by chinthakagodawita — CHANGES_REQUESTED** [review]

File: test/autoupdate.test.ts (hunk 5 above)
Comment (anarast) [reply test/autoupdate.test.ts:557]: 😳

**Review by chinthakagodawita — APPROVED** [review]
